### ISSUE TITLE
Accessibility of checkboxes, radio buttons and buttons.

### ISSUE THREAD

**Issue by blindndangerous** [AzuraCast/AzuraCast#2480]
When using azuracast with a screenreader, the checkboxes and radio buttons seem to have an extra line letting you know if it's checked or not.  I suspect this is an easy fix, as the state just seems to be reported twice.
I use a screenreader called NVDA. https://www.nvaccess.org  This allows me to navigate the computer using speech.  When looking at checkboxes and radio buttons on azuracast this is what I see.
check box  not checkedUse Web Proxy for Radio Use Web Proxy for Radio
check_box_outline_blank
The same is true for radio buttons just unchecked and checked.  If the checkbox is checked it says just checkbox instead of outline blank.  For buttons, I see a submit button and below it a save changes and close buttons at the bottom of pages.  You can see this if you go, for example to playlists>new playlist, and make one.  At the bottom their will be submit, save changes and close.

**Comment by SlvrEagle23** [comment]
@blindndangerous Hello and thank you for your feedback!

We had actually worked rather hard on making sure our checkboxes were accessible, but it appears that when we moved over to Vue components, some of those accessibility changes didn't make it over. I will investigate this and update this issue accordingly. Accessibility is a high priority for our project.

**Comment by blindndangerous** [comment]
Thank you very much, we can also see if this fixes the other issue I reported. Once this's fixed, I'll be sure to close this.  Thank you for the work you guys are doing. Running this on a pi4 (unfortunately no docker) but enjoying it all the same.

**Comment by blindndangerous** [comment]
I can open up a new issue for this if you'd like, but I've also noticed something else.
If you make a new playlist, and choose the remote URL as the source type, NVDA says Song -based playlist, below that is song playback type, an edit field which I'm guessing is for the URL, and then Remote URL Type.  If remote URL playlist is selected you can also see the text for Hide metadata from listeners checkbox's description, "Enable this setting to prevent metadata from being sent to the AutoDJ for files in this playlist. This is useful if the playlist contains jingles or bumpers.  The checkbox isn't showing up though, only that text.

**Comment by stickbear2015** [comment]
I'm running rolling release via docker, and I see that the accessibility issues mentioned herein are still valid as of the latest rolling release 
commit.

**Comment by SlvrEagle23** [comment]
@blindndangerous @stickbear2015 I've looked into this and discovered the cause of it: because we use the Material Icon set, those icons are typically just ligatures on top of text that's actually typed out, so the word "checkbox" turns into the checkbox icon; normally this is meant to be _more_ accessible for screen readers, but when it doubles up against the actual input item, clearly it doesn't serve that purpose well.

I've modified our stylesheet to set those same icons as SVG backgrounds, which the screen readers won't pick up, thus leaving only the main checkbox as a highlightable element.

That should resolve the issue on your end; if you could update and let me know if it's fixed.

**Comment by stickbear2015** [comment]
ok, so making your life another living hell with these checkboxes. :)
if you click manage, and then music files, the checkbox that appears on that page is just the select all checkbox according to my screen reader.
I can't navigate with the next checkbox command, because well, the unchecked checkboxes boxes come out as
check_box_outline_blank
check it?
I get
check_box
My screen reader doesn't actually see it as a checkbox.
Note: I did test this with NVDA
http://www.nvda-project.org
and JAWS
http://www.freedomscientific.com

**Comment by SlvrEagle23** [comment]
@stickbear2015 This should be fixed now as well.

**Comment by stickbear2015** [comment]
@SlvrEagle23 For buttons, I am still seeing a submit,  save changes and close buttons.  This is still present in most add pages, playlists, etc.
 Another area is the schedule pages.

**Comment by stickbear2015** [comment]
@SlvrEagle23 I'll test this, once I can get azuracast to successfully upgrade from 4007db1 to e1e8a7b. :) but that's an unrelated issue. :)

**Comment by stickbear2015** [comment]
@SlvrEagle23 I can confirm we've no longer got orphan buttons, I'll complete accessibility checks when I'm off work that pays me. :)

**Comment by blindndangerous** [comment]
@SlvrEagle23 Looking at the latest commit, I can move between checkboxes by using the shortcut, however, all the checkboxes say select this row regardless of if they are checked or not.  Would it be possible to have the checkboxes read the name column when we move through the table by shortcuts instead of just the text of "Select this Row".

**Comment by SlvrEagle23** [comment]
@blindndangerous We can't actually change the text of that button on a per-row basis, but we can set a row header that will hopefully be prioritized when reading the row. I've set the main "name" parameter to be the row header, so hopefully that will at least help with the context of that table.

**Comment by blindndangerous** [comment]
@SlvrEagle23 Perfect!  This looks all good from the ansible side. I'll leave this open just to make sure that @stickbear2015 has no more issues since he's on docker.  If not, I'll close this. Thank you again!

**Comment by stickbear2015** [comment]
@SlvrEagle23 @blindndangerous 
OK, we'll start with the good.
It works! :)
The checkboxes are now findable with the screen reader's quick navigation commands.
My only minor gripe is that now, their's too much screen reader chatter.
It now says
country, Select this row checkbox (insert state)
insert state being checked or unchecked.
Can we just have it say, the folder name, so speech would spit out something like
country, checkbox (insert state)
that state being checked or unchecked as earlier stated?
It's a checkbox, so I would hope that if I check this box it's gonna select that thing to what box I'm checking, so I don't need to hear Select this row. What else would it do, make me breakfast? :) lol.
Sorry, it's been a long morning already. OK, I'm not sorry, because I thought that was funny. :) 
If this isn't changeable, due to code, etc. I'll get used to it.
My rule is, I'd rather bring it up, and be told it can't be done, than not bring it up and it be fixable.

**Comment by SlvrEagle23** [comment]
@stickbear2015 @blindndangerous Well, good news and bad news on my end...

The bad news is that implementing the checkboxes as actual checkboxes in the media file manager breaks the library's built-in row selection functionality, which means that any rows selected via the checkboxes don't actually "count" as selected, so bulk actions (like deleting, assigning to a playlist, moving, etc.) don't actually take place with those files.

The only solution is to basically revert back to using a displayed checkbox that merely looks like a checkbox but isn't actually one. I've updated it to be a little more screen-reader friendly by replacing the icon names with "Select" and "Deselect", but these are just visuals indicating what clicking the row again would do.

I do notice that our library uses the `aria-selected` attribute on each table row. I'm not sure if your screen readers take that annotation into account and thus know that rows are individually selectable or not. Please let me know how the updated experience functions on your end.

**Comment by stickbear2015** [comment]
@SlvrEagle23 @blindndangerous let me update, and see what it does. I was about to pop another issue in about what you referenced about moving, etc. but you caught it before I did. :)
Let me update and check.

**Comment by stickbear2015** [comment]
@SlvrEagle23 
OK, while you can't use quick nav in your screen reader of choice, you can now tell what row's selected or de-selected.
In my humble opinion I think we've done all there is to be done with this, and it's as accessible as it can be.
Unless @blindndangerous finds something else, I think we can finally consider this closed.
Thank you for all the hard work. :)

**Comment by SlvrEagle23** [comment]
I'll go ahead and close this issue then, as the original issue reported has been resolved and I think we've taken some other important strides to ensuring the modern AzuraCast is as accessible as we can make it.

Thank you again for your patience and help as we work out these issues. If you encounter any accessibility trouble in the future, please feel free to create a new ticket.
